hack/version-matrix-gen/main.go

```diff
@@ -0,0 +1,174 @@
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	if _, err = f.WriteString(versionMatrixFileHeader); err != nil {
+		fmt.Fprintf(os.Stderr, "error writing matrix doc file header: %s\n", err)
+		os.Exit(1)
+	}
+
+	for _, version := range constellVersions {
+		kubernetesVersions := versionMatrix[version]
+		semver.Sort(kubernetesVersions)
+		k8sVersionStrings := make([]string, len(kubernetesVersions))
+		for i, kubernetesVersion := range kubernetesVersions {
+			k8sVersionStrings[i] = kubernetesVersion.String()
+		}
+		if _, err := f.WriteString(
+			fmt.Sprintf("| %s | %s |\n", version, strings.Join(k8sVersionStrings, "<br></br>")),
+		); err != nil {
+			fmt.Fprintf(os.Stderr, "error writing matrix doc file: %s\n", err)
+			os.Exit(1)
+		}
+	}
+}
+
+// readVersionMatrix reads the version matrix doc file and returns a map of
+// supported Constellation versions to supported Kubernetes versions.
+func readVersionMatrix(path string) (map[semver.Semver][]semver.Semver, error) {
+	fileContent, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	cleanedFileContent := strings.TrimSpace(string(fileContent))
+
+	// Split the file content to extract the version matrix
+	// We want all content after the table header (starting with "|-")
+	lines := strings.Split(cleanedFileContent, "\n")
+	var matrixLines []string
+	for i, line := range lines {
+		if strings.HasPrefix(line, "|-") {
+			if i+1 >= len(lines) {
+				return nil, fmt.Errorf("invalid version matrix: no content after table header")
+			}
+			matrixLines = lines[i+1:]
+			break
+		}
+	}
+
+	// Parse the version matrix.
+	matrix := make(map[semver.Semver][]semver.Semver, len(matrixLines))
+	for _, line := range matrixLines {
+		// Split the line into the Constellation version and the supported
+		// Kubernetes versions.
+		fields := strings.Split(
+			strings.Trim(
+				strings.ReplaceAll(line, " ", ""),
+				"|",
+			),
+			"|",
+		)
+		if len(fields) != 2 {
+			return nil, fmt.Errorf("invalid version matrix line: %q", line)
+		}
+
+		constellationVersion, err := semver.New(fields[0])
+		if err != nil {
+			return nil, fmt.Errorf("invalid constellation version in version matrix line: %q", line)
+		}
+
+		kubernetesVersionStrings := strings.Split(fields[1], "<br></br>")
+		kubernetesVersions := make([]semver.Semver, len(kubernetesVersionStrings))
+		for i, kubernetesVersionString := range kubernetesVersionStrings {
+			kubernetesVersions[i], err = semver.New(kubernetesVersionString)
+			if err != nil {
+				return nil, fmt.Errorf("invalid kubernetes versions in version matrix line: %q", line)
+			}
+		}
+		matrix[constellationVersion] = kubernetesVersions
+	}
+
+	return matrix, nil
+}
+
+func parseNewVersion(cliPath string) (semver.Semver, []semver.Semver, error) {
+	// Run the CLI to get
+	// 1. the new version
+	// 2. the supported Kubernetes versions
+
+	out, err := exec.CommandContext(context.Background(), cliPath, "version").Output()
```

Comment:
Wouldn't an easier way to just import the `versions.go` ? 
Also we could add the matrix to `bazel run //:generate` to verify that everything has the correct state.

Reply:
Not easier, because we would need a lot of extra checks to make sure we dont update the matrix with any pseudo versions.
Also need to do some changes to the release job that updates the docs, since this creates a branch from main, not from the pending release branch.

I could see a way to make this work though, and it could also make it easier to update the matrix for patch releases.